templates/main.tftpl

```diff
@@ -1 +1,6 @@
-#This file is where the code for your application will be in the future.
\ No newline at end of file
+%{ for rg, rg_name in rg_config }resource "azurerm_resource_group" "${rg}" {
+  name     = "rg-$${local.application_name}${rg_name}-$${local.stage}-$${local.env_num}"
```

Comment:
Accessing locals directly is not possible. You need to pass them in repository.tf